packages/synapse-react-client/src/components/SynapseTable/export/ExternalPlatformActionsRequiredPrecheck.test.tsx

```diff
@@ -0,0 +1,374 @@
+import {
+  ColumnModel,
+  ColumnSingleValueFilterOperator,
+  ColumnTypeEnum,
+  QueryBundleRequest,
+  SelectColumn,
+} from '@sage-bionetworks/synapse-types'
+import { act, render, screen, waitFor } from '@testing-library/react'
+import { upperFirst } from 'lodash-es'
+import { useExportTableQueryToAnalysisPlatform } from '../../../synapse-queries/entity/useExportTableQueryToAnalysisPlatform'
+import { getUseQuerySuccessMock } from '../../../testutils/ReactQueryMockUtils'
+import { TableQueryActionsRequired } from '../../DownloadCart/TableQueryActionsRequired'
+import { useQueryContext } from '../../QueryContext'
+import { QueryContextType } from '../../QueryContext/index'
+import {
+  QueryVisualizationContextType,
+  useQueryVisualizationContext,
+} from '../../QueryVisualizationWrapper/index'
+import {
+  useHasSelectedRowsAtomValue,
+  useRowSelectionPrimaryKeyAtomValue,
+  useSelectedRowsAtomValue,
+} from '../../QueryWrapper/TableRowSelectionState'
+import { useGetQueryMetadata } from '../../QueryWrapper/useGetQueryMetadata'
+import {
+  EXTERNAL_ANALYSIS_PLATFORMS,
+  ExternalAnalysisPlatform,
+} from './ExternalAnalysisPlatformsConstants'
+import ExternalPlatformActionsRequiredPrecheck from './ExternalPlatformActionsRequiredPrecheck'
+
+jest.mock(
+  '../../../synapse-queries/entity/useExportTableQueryToAnalysisPlatform',
+)
+jest.mock('../../DownloadCart/TableQueryActionsRequired')
+jest.mock('../../QueryContext')
+jest.mock('../../QueryWrapper/useGetQueryMetadata')
+jest.mock('../../QueryVisualizationWrapper/index')
+jest.mock('../../QueryWrapper/TableRowSelectionState')
+
+const mockUseExportTableQueryToAnalysisPlatform = jest.mocked(
+  useExportTableQueryToAnalysisPlatform,
+)
+const mockUseQueryContext = jest.mocked(useQueryContext)
+const mockUseQueryVisualizationContext = jest.mocked(
+  useQueryVisualizationContext,
+)
+const mockUseGetQueryMetadata = jest.mocked(useGetQueryMetadata)
+const MockTableQueryActionsRequired = jest.mocked(TableQueryActionsRequired)
+const mockOnConfirmButtonPropsChange = jest.fn()
+const mockOnSuccessfulExport = jest.fn()
+
+const mockUseHasSelectedRowsAtomValue = jest.mocked(useHasSelectedRowsAtomValue)
+const mockUseRowSelectionPrimaryKeyAtomValue = jest.mocked(
+  useRowSelectionPrimaryKeyAtomValue,
+)
+const mockUseSelectedRowsAtomValue = jest.mocked(useSelectedRowsAtomValue)
+
+const mockFileIdColumnName = 'file_id'
+const mockFileVersionColumnName = 'file_version'
+const mockFileNameColumnName = 'file_name'
+
+const queryRequest: QueryBundleRequest = {
+  concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+  query: {
+    sql: 'SELECT * FROM syn123',
+  },
+  entityId: 'syn123',
+  partMask: 1,
+}
+
+const columnModels: ColumnModel[] = [
+  { id: '123', name: 'foo', columnType: ColumnTypeEnum.STRING },
+]
+const selectColumns: SelectColumn[] = columnModels
+
+const mockQueryContext: QueryContextType = {
```

Comment:
I'm surprised you had to define all of these (that we don't have existing mock objects that are reused across tests)